### PR TITLE
Adding ID Prefix Reservation docs

### PR DESCRIPTION
**Pull request by diverdan92**

### CODE REVIEW

**Comment by saldana** [comment]
### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Reference/ID-Prefix-Reservation.md](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/ID-Prefix-Reservation.md) | :warning:Warning | | [Details](#user-content-docs%2FReference%2FID-Prefix-Reservation.md)
[docs/Reference/media/nuget-gallery-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/nuget-gallery-reserved-prefix.png) | :white_check_mark:Succeeded | |
[docs/Reference/media/visual-studio-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/visual-studio-reserved-prefix.png) | :white_check_mark:Succeeded | |

<a name="docs%2FReference%2FID-Prefix-Reservation.md"></a>
### [docs/Reference/ID-Prefix-Reservation.md](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/ID-Prefix-Reservation.md)
 - **[Warning]** Author dajaco cannot be recognized in GitHub.

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C10%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710100029476740-501%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=Mq9cjvukOtH77X0uKSBd3KoF4zQjIDeJLmyMoMBXK58%3D&st=2017-10-10T00%3A25%3A30Z&se=2017-11-10T00%3A30%3A30Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by diverdan92** [comment]
@unniravindranathan, @PatoBeltran, @karann-msft, @anangaur, @shishirx34 - can you review?

**Comment by saldana** [comment]
### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Reference/ID-Prefix-Reservation.md](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/ID-Prefix-Reservation.md) | :warning:Warning | | [Details](#user-content-docs%2FReference%2FID-Prefix-Reservation.md)
[docs/Reference/media/nuget-gallery-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/nuget-gallery-reserved-prefix.png) | :white_check_mark:Succeeded | |
[docs/Reference/media/visual-studio-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/visual-studio-reserved-prefix.png) | :white_check_mark:Succeeded | |

<a name="docs%2FReference%2FID-Prefix-Reservation.md"></a>
### [docs/Reference/ID-Prefix-Reservation.md](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/ID-Prefix-Reservation.md)
 - **[Warning]** Author dajaco cannot be recognized in GitHub.

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C10%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710100059358306-501%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=LnYE8ia7F8HxdxAPV55xS8A5PKalvcnS1Ex3jXX54Zs%3D&st=2017-10-10T01%3A06%3A52Z&se=2017-11-10T01%3A11%3A52Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/Reference/ID-Prefix-Reservation.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+- karann
+- unnir
+#ms.suite:
+#ms.tgt_pltfrm:
+#ms.custom:
+
+---
+
+# Package ID prefix reservation
+[NuGet.org](https://www.nuget.org/) and Visual Studio 2017 version 15.4 or later now show a visual indicator for packages that are submitted by owners with a reserved package ID prefix, as long as the package matches the reserved ID prefix naming pattern. The below reference explains what the ID prefix reservation entails, and how an owner can apply for an ID prefix.
```
Comment (karann-msft) [inline docs/Reference/ID-Prefix-Reservation.md]: Docs shouldn't have a point in time references such as "now"

File: docs/Reference/ID-Prefix-Reservation.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+# optional metadata
+
+description: Package ID Prefix Reservation feature description and author guide.
+keywords: NuGet package ID, prefix, reservation
+#ROBOTS:
+#audience:
+#ms.devlang:
+ms.reviewer:
+- anangaur
```
Comment (karann-msft) [inline docs/Reference/ID-Prefix-Reservation.md]: use github userid for all of us, for example, use karann-msft for me

File: docs/Reference/ID-Prefix-Reservation.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+#ms.tgt_pltfrm:
+#ms.custom:
+
+---
+
+# Package ID prefix reservation
+[NuGet.org](https://www.nuget.org/) and Visual Studio 2017 version 15.4 or later now show a visual indicator for packages that are submitted by owners with a reserved package ID prefix, as long as the package matches the reserved ID prefix naming pattern. The below reference explains what the ID prefix reservation entails, and how an owner can apply for an ID prefix.
+
+## What is package ID prefix reservation?
+We've heard from the NuGet community that there are identity issues when trying to consume NuGet packages. We've also heard from NuGet package owners that sometimes they are forced to make takedown requests for packages that are squatting on their ID prefixes - something that can cause delays in shipping new packages. To alleviate pain in both the package consumer and package owner scenarios, we have built a package ID prefix reservation feature.
```
Comment (karann-msft) [inline docs/Reference/ID-Prefix-Reservation.md]: This sounds like a blog post, I don't thing we need to explain the reason why we are implementing this feature. Sure, we can call out the advantages,

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Reference/ID-Prefix-Reservation.md](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/ID-Prefix-Reservation.md) | :white_check_mark:Succeeded | |
[docs/Reference/media/nuget-gallery-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/nuget-gallery-reserved-prefix.png) | :white_check_mark:Succeeded | |
[docs/Reference/media/visual-studio-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/visual-studio-reserved-prefix.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C10%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710101730496895-501%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=EmWonpBuGSrARBiPEmVqWyK3O%2FG7%2BjzBePzlUYjlgCg%3D&st=2017-10-10T17%3A28%3A08Z&se=2017-11-10T17%3A33%3A08Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Reference/ID-Prefix-Reservation.md](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/ID-Prefix-Reservation.md) | :white_check_mark:Succeeded | |
[docs/Reference/media/nuget-gallery-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/nuget-gallery-reserved-prefix.png) | :white_check_mark:Succeeded | |
[docs/Reference/media/visual-studio-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/visual-studio-reserved-prefix.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C10%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710101734050601-501%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=JPl0x8ISPEopRv3WFSwo6ezKbs1a8SEWmuTPgJ1DaEY%3D&st=2017-10-10T17%3A31%3A06Z&se=2017-11-10T17%3A36%3A06Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/Reference/ID-Prefix-Reservation.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+4. All previously existing packages that match the reserved ID prefix, but are *not* owned by the owner of the reserved prefix will remain unchanged (they will not be unlisted, but they will also not have the visual indicator).
+
+These changes are based on the following conditions and impose several additional restrictions:
+* Only one owner of a package needs to have the reserved prefix for the visual indicator to appear (for packages with multiple-owners).
+* If there is more than one owner of a package where one or more owners has the reserved prefix and one or more owners does not have the reserved prefix, then only the owner(s) with the reserved prefix can remove other owner(s) with a reserved prefix. The owners who do not have the prefix reserved cannot remove owners with the prefix reserved. They can still remove other owners that also do not have the prefix reserved.
+  * Once a package has the visual indicator, it should *always* have the visual indicator (guarunteeing that at least one owner with the reserved prefix will always remain an owner)
```
Comment (shishirx34) [inline docs/Reference/ID-Prefix-Reservation.md]: guaranteeing

Comment (diverdan92) [reply docs/Reference/ID-Prefix-Reservation.md]: Fixed!

File: docs/Reference/ID-Prefix-Reservation.md (hunk 3 above)
Comment (diverdan92) [reply docs/Reference/ID-Prefix-Reservation.md]: Updated

File: docs/Reference/ID-Prefix-Reservation.md (hunk 2 above)
Comment (diverdan92) [reply docs/Reference/ID-Prefix-Reservation.md]: Updated

File: docs/Reference/ID-Prefix-Reservation.md (hunk 1 above)
Comment (diverdan92) [reply docs/Reference/ID-Prefix-Reservation.md]: Updated

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Reference/ID-Prefix-Reservation.md](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/ID-Prefix-Reservation.md) | :white_check_mark:Succeeded | |
[docs/Reference/media/nuget-gallery-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/nuget-gallery-reserved-prefix.png) | :white_check_mark:Succeeded | |
[docs/Reference/media/visual-studio-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/visual-studio-reserved-prefix.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C10%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710101749188287-501%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=oNH3T%2Fe2gC7tIeV6%2F%2BrwD0hoZLlLiM10o42Om6ppfz8%3D&st=2017-10-10T17%3A46%3A19Z&se=2017-11-10T17%3A51%3A19Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/Reference/ID-Prefix-Reservation.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+1. Does the package ID prefix properly and clearly identify the package owner?
+2. Are a significant number of the packages that have already been submitted by the owner under the package ID prefix?
+3. Is the package ID prefix something common that should not belong to any individual owner or organization?
+4. Would *not* reserving the package ID prefix cause ambiguity and confusion for the community?
+5. Are the identifying properties of the packages that match the package ID prefix clear and consistent (especially the package author)?
+
+## 3rd party feed provider scenarios
+If a 3rd party feed provider is interested in implementing their own service to provide prefix reservations, you can do so by modifying the search service in the NuGet v2 or v3 feed providers. The addition in the feed search service is to add the *verified* property, with examples for both the v2 and v3 feeds below.
```
Comment (shishirx34) [inline docs/Reference/ID-Prefix-Reservation.md]: I checked with Patricio and the client supports the prefix reservation property on the v3 search feed only. You can get rid of the v2 feed changes from here and below.

File: docs/Reference/ID-Prefix-Reservation.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+### ID prefix reservation on NuGet.org
+When a prefix is reserved on [NuGet.org](https://www.nuget.org/), the following will happen:
+1. A prefix reservation is associated with an owner or set of owners on [NuGet.org](https://www.nuget.org/). 
+2. Whenever a package is submitted to [NuGet.org](https://www.nuget.org/) with an ID that matches the reserved ID prefix, the package is rejected unless it originates from the owner(s) that reserved the ID prefix.
+3. Any package that matches the reserved ID prefix and originates from the owner(s) that reserved the ID prefix will have a visual indicator in Visual Studio 2017 version 15.4 or later, and on [NuGet.org](https://www.nuget.org/) indicating that the package is under a reserved ID prefix. This is true for both new package submissions as well as existing packages under the owner(s). **Note:** The indicator in Visual Studio will only appear if a single feed is selected as the package source. 
+4. All previously existing packages that match the reserved ID prefix, but are *not* owned by the owner of the reserved prefix will remain unchanged (they will not be unlisted, but they will also not have the visual indicator).
```
Comment (shishirx34) [inline docs/Reference/ID-Prefix-Reservation.md]: Should we mention that the existing packages can still upload new versions here or somewhere?

File: docs/Reference/ID-Prefix-Reservation.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+### ID prefix reservation criteria
+When reviewing any application for ID prefix reservation, the [NuGet.org](https://www.nuget.org/) team will evalaute the application against the below criteria. Not all criteria needs to be met for a prefix to be reserved, but the application may be denied if there is not substantial evidence of the criteria being met (with an explanation given):
+1. Does the package ID prefix properly and clearly identify the package owner?
+2. Are a significant number of the packages that have already been submitted by the owner under the package ID prefix?
+3. Is the package ID prefix something common that should not belong to any individual owner or organization?
+4. Would *not* reserving the package ID prefix cause ambiguity and confusion for the community?
+5. Are the identifying properties of the packages that match the package ID prefix clear and consistent (especially the package author)?
+
+## 3rd party feed provider scenarios
```
Comment (shishirx34) [inline docs/Reference/ID-Prefix-Reservation.md]: what about the guidelines to allocate namespace reservation? Shouldn't we suggest some key points for a 3rd party feed to assign/allocate a reserved prefix, similar adherence to our guidelines or something like that?

File: docs/Reference/ID-Prefix-Reservation.md (hunk 5 above)
Comment (diverdan92) [reply docs/Reference/ID-Prefix-Reservation.md]: Updated

File: docs/Reference/ID-Prefix-Reservation.md (hunk 6 above)
Comment (diverdan92) [reply docs/Reference/ID-Prefix-Reservation.md]: Updated

File: docs/Reference/ID-Prefix-Reservation.md (hunk 7 above)
Comment (diverdan92) [reply docs/Reference/ID-Prefix-Reservation.md]: What do you have in mind? Independent feed providers can choose how they want to use the feature, such as reserving for specific teams, reserving for internally created packages, or similar?

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Reference/ID-Prefix-Reservation.md](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/ID-Prefix-Reservation.md) | :white_check_mark:Succeeded | |
[docs/Reference/media/nuget-gallery-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/nuget-gallery-reserved-prefix.png) | :white_check_mark:Succeeded | |
[docs/Reference/media/visual-studio-reserved-prefix.png](https://github.com/diverdan92/docs.microsoft.com-nuget/blob/patch-2/docs/Reference/media/visual-studio-reserved-prefix.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C10%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710101856508305-501%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=kTvK%2Boo7MUoH427HoWReI%2B2SY1bJEOnSRebkvhpyjCc%3D&st=2017-10-10T18%3A53%3A01Z&se=2017-11-10T18%3A58%3A01Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**